_protected/app/system/modules/mail/views/base/tpl/main/msg.inc.tpl

```diff
@@ -14,19 +23,56 @@
   {{ $set_to = ($is_delete) ? 'setdelete' : 'settrash' }}
   {{ $label_txt = ($is_delete) ? t('Delete') : t('Trash') }}
 
-  <div class="center">
-    <dl>
-      <dt>{lang 'Author:'}</dt>
-      <dd>{{ $avatarDesign->get($usernameSender, $firstNameSender, null, 32) }}</dd>
-      <dt>{lang 'Subject:'}</dt>
-      <dd>{subject}</dd>
-      <dt>{lang 'Message:'}</dt>
-      <dd>{message}</dd>
-      <dt>{lang 'When:'}</dt>
-      <dd>{% Framework\Date\Various::textTimeStamp($msg->sendDate) %}</dd>
-    </dl>
+  {{ $subject = escape(str_replace('re ', '', $msg->title), true) }}
+
+  <div class="left">
+  <dl>
+    <dt class="msg-label">{lang 'From:'}</dt>
+    <dd>{{ $avatarDesign->get($usernameSender, $firstNameSender, null, 32) }}</dd>
+    <dd class="msg-from"><?php echo $usernameSender ?></dd>
+    <dt class="msg-label">{lang 'Subject:'}</dt>
+    <dd class="msg-cc">{subject}</dd>
+  </dl>
+    
+  {if !empty($msgsOld)}
+           
+  <h6>Previous Messages:</h6>
+
+  {each $msg in $msgsOld}
+  
+  {* Set Variables *}
+  {{ $usernameSender = (empty($msg->username)) ? PH7_ADMIN_USERNAME : escape($msg->username) }}
+  {{ $firstNameSender = (empty($msg->firstName)) ? PH7_ADMIN_USERNAME : escape($msg->firstName) }}
+  {{ $subject = escape(Framework\Security\Ban\Ban::filterWord($msg->title)) }}
+  {{ $message = Framework\Parse\Emoticon::init(Framework\Security\Ban\Ban::filterWord($msg->message)) }}
+  {{ $is_outbox = ($msg->sender == $member_id) }}
+  {{ $is_trash = (($msg->sender == $member_id && $msg->trash == 'sender') || ($msg->recipient == $member_id && $msg->trash == 'recipient') && !$is_outbox) }}
+  {{ $is_delete = ($is_outbox || $is_trash) }}
+  {{ $set_to = 'setdelete' }}
+  {{ $label_txt = t('Delete') }}
+  
+  {{ $subject = escape(str_replace('re ', '', $msg->title), true) }}
+    
+      <!-- Sent by User -->
+      <div class="msg-left">
+        <dl>
+          <dd class="msg-text-area">{message}</dd>
+           <div>{{ LinkCoreForm::display($label_txt, 'mail', 'main', $set_to, array('id'=>$msg->messageId)) }}
+          {if $is_trash} | {{ LinkCoreForm::display(t('Move to Inbox'), 'mail', 'main', 'setrestore', array('id'=>$msg->messageId)) }}{/if}</div>
+          {if $msg->sender === $member_id}
+          <dt class="msg-l-sent">Sent: &nbsp;{% Framework\Date\Various::textTimeStamp($msg->sendDate) %}</dt>
+          {else}
+          <dt class="msg-l-received">Received: &nbsp;{% Framework\Date\Various::textTimeStamp($msg->sendDate) %}</dt>
+          {/if}
+        </dl>
+        
+    {/each}
+    {/if}
+        <hr>
+       
 
-    <div><a href="{{ $design->url('mail','main','compose',"$usernameSender,$subject") }}">{lang 'Reply'}</a> | {{ LinkCoreForm::display($label_txt, 'mail', 'main', $set_to, array('id'=>$msg->messageId)) }}
-    {if $is_trash} | {{ LinkCoreForm::display(t('Move to Inbox'), 'mail', 'main', 'setrestore', array('id'=>$msg->messageId)) }}{/if}</div>
-  </div>
+        {{ MailForm::display() }}
```

Comment:
We will probably have to change that form. Otherwise, we get the recipient and subject fields as well when we reply.

Reply:
Yeah I couldn't pass the username of the sender or the subject to the form. I think its because its being called before the functions in the controller then the model have had enough time to return the values. The form is there instantly the rest of the process is still taking place.

I just noticed if you are the sender you can delete messages from the list but if you are recipient it shows move to inbox. I will have a look when I'm able to.

Did you like the idea of the search being there on the page? I have done this with photo's and Video's too on my live site.

Reply:
How would you change the form? Remove the recipient and subject fields as a reply should hold the subject with one Re: preceding it but I don't see how the info can be added before the form is called? Maybe in the model is the ideal place?